assimilation_code/modules/assimilation/filter_mod.f90

```diff
@@ -2335,12 +2339,11 @@ function count_state_ens_copies(ens_size, post_inflate, prior_inflate) result(nu
                        PRIOR_INF_COPY, PRIOR_INF_SD_COPY, POST_INF_COPY, POST_INF_SD_COPY /)
 
 ! CSS If Whitaker/Hamill (2012) relaxation-to-prior-spread (rpts) inflation (inf_flavor = 4)
-!  then we need an extra copy to hold (save) the prior ensemble spread
-!   ENS_SD_COPY will be overwritten with the posterior spread before
-!   applying the inflation algorithm; hence we must save the prior ensemble spread in a different copy
-if ( inf_flavor(2) == 4 ) then ! CSS
-   SPARE_PRIOR_SPREAD = next_copy_number(cnum)
-endif 
+! then we need an extra copy to hold (save) the prior ensemble spread
+! ENS_SD_COPY will be overwritten with the posterior spread before
+! applying the inflation algorithm; hence we must save the prior ensemble spread in a different copy
+
+if (do_rtps_inflate(post_inflate)) SPARE_PRIOR_SPREAD = next_copy_number(cnum)
```

Comment:
This is .FALSE. even though      inf_flavor(2) == 4

Given the code in the previous part of the routine, I thought this should work. I do not know why this is not reproducing.
This is NOT WORKING CORRECTLY.